azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java

```diff
@@ -1634,6 +1758,52 @@ public StorageAccount call(StorageAccount storageAccount) {
         return Observable.just(null);
     }
 
+    private Observable<StorageAccount> handleBootDiagnosticsStorageSettings(StorageAccount diskStorageAccount) {
+        final Observable<StorageAccount> diskStgObservable = Observable.just(diskStorageAccount);
+        if (this.inner().diagnosticsProfile() == null
```

Comment:
for multiple ifs getting diagnostic profile from inner can be a costly transaction. It would be better to get diagnostic profile and then to perform null checks onto it.